agent/plugin/error.go

```diff
@@ -0,0 +1,123 @@
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+type unit struct{}
+
+// DeprecatedNameErrors contains a set of DeprecatedNameError
+type DeprecatedNameErrors struct {
+	errs map[DeprecatedNameError]unit
+}
+
+// IsEmpty return true if and only if `e` contains no errors
+func (e *DeprecatedNameErrors) IsEmpty() bool {
+	return e == nil || len(e.errs) == 0
+}
+
+// Errors returns the contained set of errors in sorted order
+func (e *DeprecatedNameErrors) Errors() []DeprecatedNameError {
+	if e == nil {
+		return []DeprecatedNameError{}
```

Comment:
Why not `return nil`?

Reply:
Yep, and I made it handle when `e.errs` is `nil` too.